Makefile

```
@@ -195,6 +197,73 @@ bundle: manifests kustomize
 	sed -i "/^annotations.*/a \  # Annotations for RedHat Openshift Platform" bundle/metadata/annotations.yaml; \
 
 
+.PHONY: bundle-operatorhub
+bundle-operatorhub: manifests kustomize
+	$(eval BUNDLE_DIR:= $(ROOT_DIR)/bundle/$(DISTRIBUTION)/$(VERSION)/)
+	$(eval ANNOTATIONS_FILE_PATH:= $(BUNDLE_DIR)/metadata/annotations.yaml)
+	$(eval KUSTOMIZE_DIR:= $(OVERLAYS_DIR)/base)
+
+	if [ -f $(ROOT_DIR)/bundle.Dockerfile ]; then \
+        rm -f $(ROOT_DIR)/bundle.Dockerfile; \
+      fi; \
+    if [ -d $(BUNDLE_DIR) ]; then \
+      rm -r $(BUNDLE_DIR); \
+      fi; \
+	operator-sdk generate kustomize manifests -q
+	cd $(ROOT_DIR)/config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
+	cd $(ROOT_DIR)/config/manifests/bases && $(KUSTOMIZE) edit set annotation containerImage:$(IMG) createdAt:$(DATE)
+	cd "${ROOT_DIR}" && "${KUSTOMIZE}" build "${KUSTOMIZE_DIR}" | \
+    operator-sdk generate bundle --overwrite --version $(VERSION) --output-dir $(BUNDLE_DIR) $(BUNDLE_METADATA_OPTS); \
+
+	if [ $(DISTRIBUTION) == operatorhub ]; then \
+        operator-sdk bundle validate $(BUNDLE_DIR) --select-optional name=$(DISTRIBUTION); \
+    else \
+      sed -i "/^annotations.*/a \  com.redhat.openshift.versions: $(OPENSHIFT_VERSION)" $(ANNOTATIONS_FILE_PATH); \
+      sed -i "/^annotations.*/a \  # Annotations for RedHat Openshift Platform" $(ANNOTATIONS_FILE_PATH); \
+      sed -i "/^FROM.*/a LABEL com.redhat.openshift.versions=$(OPENSHIFT_VERSION)" $(ROOT_DIR)/bundle.Dockerfile; \
+      sed -i "/^FROM.*/a LABEL com.redhat.delivery.operator.bundle=true" $(ROOT_DIR)/bundle.Dockerfile; \
+      sed -i "/^FROM.*/a LABEL com.redhat.delivery.backport=false" $(ROOT_DIR)/bundle.Dockerfile; \
+      sed -i "/^FROM.*/a # Labels for RedHat Openshift Platform" $(ROOT_DIR)/bundle.Dockerfile; \
+      fi; \
+
+
+.PHONY: bundle-ocp
+bundle-ocp: DISTRIBUTION=ocp
+bundle-ocp: bundle-operatorhub
+
+.PHONY: bundle-rhmp
+bundle-rhmp: DISTRIBUTION=rhmp
+bundle-rhmp: manifests kustomize
+	$(eval BUNDLE_DIR:= $(ROOT_DIR)/bundle/$(DISTRIBUTION)/$(VERSION)/)
+	$(eval ANNOTATIONS_FILE_PATH:= $(BUNDLE_DIR)/metadata/annotations.yaml)
+	$(eval KUSTOMIZE_DIR:= $(OVERLAYS_DIR)/$(DISTRIBUTION))
+	$(eval BUNDLE_METADATA_OPTS:= $(BUNDLE_METADATA_OPTS) --use-image-digests)
+
```

```suggestion
	if [ -f $(ROOT_DIR)/bundle.Dockerfile ]; then \
        rm -f $(ROOT_DIR)/bundle.Dockerfile; \
      fi; \
    if [ -d $(BUNDLE_DIR) ]; then \
      rm -r $(BUNDLE_DIR); \
      fi; \
```
can be replaced with 
```
rm -rf bundle.Dockerfile bundle/
```
This is the same as we do in
```
.PHONY: bundle
bundle: manifests kustomize
```